spec/classes/puppet_server_spec.rb

```diff
@@ -1,181 +1,754 @@
+            .with_group('puppet') \
+            .with_mode('0750')
+
+          should contain_file("#{ssldir}/private_keys/puppetmaster.example.com.pem") \
+            .with_group('puppet') \
+            .with_mode('0640')
+
+          should contain_exec('puppet_server_config-create_ssl_dir') \
+            .with_creates(ssldir) \
+            .with_command("/bin/mkdir -p #{ssldir}") \
+            .with_umask('0022')
+
+          should contain_exec('puppet_server_config-generate_ca_cert') \
+            .with_creates("#{ssldir}/certs/puppetmaster.example.com.pem") \
+            .with_command("#{puppetcacmd} --generate puppetmaster.example.com --allow-dns-alt-names") \
+            .with_umask('0022') \
+            .that_requires(["Concat[#{conf_file}]", 'Exec[puppet_server_config-create_ssl_dir]'])
+        end
+
+        it { should contain_puppet__config__main('environmentpath').with_value(environments_dir) }
+        it { should contain_exec('puppet_server_config-generate_ca_cert').that_notifies('Service[puppetserver]') }
+
+        it 'should set up the environments' do
+          should contain_file(environments_dir)
+            .with_ensure('directory')
+            .with_owner('puppet')
+            .with_group(nil)
+            .with_mode('0755')
+
+          should contain_file(sharedir).with_ensure('directory')
+
+          should contain_file("#{codedir}/environments/common")
+            .with_ensure('directory')
+            .with_owner('puppet')
+            .with_group(nil)
+            .with_mode('0755')
+
+          should contain_file("#{sharedir}/modules")
+            .with_ensure('directory')
+            .with_owner('puppet')
+            .with_group(nil)
+            .with_mode('0755')
+
+          should contain_puppet__server__env('development')
+          should contain_puppet__server__env('production')
+        end
+
+        it { should contain_concat(conf_file) }
+
+        it { should_not contain_puppet__config__agent('configtimeout') }
+        it { should_not contain_class('puppetdb') }
+        it { should_not contain_class('puppetdb::master::config') }
+        it { should_not contain_file("#{confdir}/custom_trusted_oid_mapping.yaml") }
+
+        it { should contain_file("#{confdir}/autosign.conf") }
+        it { should_not contain_file("#{confdir}/autosign.conf").with_content(/# Managed by Puppet/) }
+        it { should_not contain_file("#{confdir}/autosign.conf").with_content(/foo.bar/) }
+
+        it 'should set up the ENC' do
+          should contain_class('foreman::puppetmaster')
+            .with_foreman_url('https://foo.example.com')
+            .with_receive_facts(true)
+            .with_puppet_home(puppetserver_vardir)
+            .with_puppet_etcdir(etcdir)
+            .with_timeout(60)
+          # Since this is managed inside the foreman module it does not
+          # make sense to test it here
+          # with_puppet_basedir('/usr/lib/ruby/site_ruby/1.9/puppet').
```

Comment:
leftover?

Reply:
Updated to verify it passes in the correct dir.